src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -311,6 +312,15 @@
    jruby-request/wrap-with-error-handling
    ring/wrap-params))
 
+(schema/defn ^:always-validate valid-static-file-path?
+  "Helper function to decide if a static_file_content path is valid.
+  The access here is designed to mimic Puppet's file_content endpoint."
+  [path :- schema/Str]
+  (when-let [canonicalized-path (URIUtil/canonicalPath path)]
+    (or
+     (bidi.bidi/match-route [["modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path)
+     (bidi.bidi/match-route [["environments/" :environment-name "/modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path))))
```

Comment:
Hmm...is this line needed? An agent will always be hitting the `/static_file_content` and passing in a code-id and an environment, so I don't think it should ever be trying to access anything in the `environments` directory directly through the URI path.